starter-apps/vue/vue-template/src/views/ReportABug.vue

```diff
@@ -0,0 +1,204 @@
+import Text from '@/components/Text.vue'
+import Input from '@/components/forms/Input.vue'
+import TextArea from '@/components/forms/TextArea.vue'
+import Button from '@/components/forms/Button.vue'
+import DateModified from '@/components/DateModified.vue'
+
+const { t } = useI18n()
+
+/*
+ * TODO: Add your form elements here. You may use the formElements object to store the form element IDs.
+ * The v-models uses the formStore to store the form data.
+ */
+const formStore = useFormStore()
+const formData = ref({ ...formStore.formData })
+const submitted = ref(formStore.submitted)
+
+const formElements = {
+  version: 'version',
+  title: 'title',
+  currentBehavior: 'currentBehavior',
+  expectedBehavior: 'expectedBehavior',
+  systemInfo: 'systemInfo',
+  stepsToReproduce: 'stepsToReproduce',
+  codeReproductionUrl: 'codeReproductionUrl',
+  additionalInfo: 'additionalInfo'
+}
+
+let errors = []
+
+const handleError = (event) => {
+  if (!errors.includes(event.detail.id)) {
+    errors.push(event.detail.id)
+  }
+}
+const handleValid = (event) => {
+  if (errors.includes(event.detail.id)) {
+    errors = errors.filter((error) => error !== event.detail.id)
+  }
+}
+async function handleSubmit() {
+  setTimeout(() => {
+    // Check if there are any errors before submitting the form
+    if (errors.length === 0) {
+      formStore.submitForm(formData.value)
+      submitted.value = formStore.submitted
+    }
+  }, 50)
+}
+
+// Link to the GitHub issue form with the form data pre-filled
+const githubIssueURL = () => {
+  const version = encodeURIComponent(formData.value.version)
+  const title = encodeURIComponent(formData.value.title)
+  const currentBehavior = encodeURIComponent(formData.value.currentBehavior)
+  const expectedBehavior = encodeURIComponent(formData.value.expectedBehavior)
+  const systemInfo = encodeURIComponent(formData.value.systemInfo)
+  const stepsToReproduce = encodeURIComponent(formData.value.stepsToReproduce)
+  const codeReproductionUrl = encodeURIComponent(formData.value.codeReproductionUrl)
+  const additionalInfo = encodeURIComponent(formData.value.additionalInfo)
+
+  const url =
+    `https://github.com/cds-snc/gcds-components/issues/new` +
+    `?=&template=bug_report.yml` +
+    `&title=${title}` +
+    `&package_version=${version}` +
+    `&current_behavior=${currentBehavior}` +
+    `&expected_behavior=${expectedBehavior}` +
+    `&sys_info=${systemInfo}` +
+    `&steps_to_reproduce=${stepsToReproduce}` +
+    `&code_url=${codeReproductionUrl}` +
+    `&more_info=${additionalInfo}`
+
+  return url
+}
+</script>
+<template>
+  <div>
+    <section>
+      <Heading tag="h1">{{ t('reportABugPage.heading') }}</Heading>
+      <Text>{{ t('reportABugPage.intro') }}</Text>
+    </section>
+    <form
+      v-if="!submitted"
+      :key="reportABugKey"
+      name="bugReportForm"
+      @gcdsError="handleError"
+      @gcdsValid="handleValid"
+      @submit.prevent="handleSubmit"
+    >
+      <Input
```

Comment:
Could we add an error summary into this form since it is a bigger form? You could then switch each form component to validate on submit.